recipes/denoptim/meta.yaml

```
@@ -0,0 +1,39 @@
+{% set version = "3.2.1" %}
+{% set openjdk_version = "11.0.*" %}
+
+package:
+  name: denoptim
+  version: {{ version }}
+
+source:
+  url: https://github.com/denoptim-project/DENOPTIM/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 1c896a412ce5d91f8fa9069ce982e90987becdf9b54514f30bfcf60a8fdc67da
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - openjdk =={{ openjdk_version }}
+    - maven
+  run:
+    - openjdk =={{ openjdk_version }}
+    - python
+    - py4j
+
+test:
+  commands:
+    - denoptim -v
+
+about:
+  home: https://github.com/denoptim-project/DENOPTIM
+  license: AGPL-3.0-only
+  license_file: COPYING
```

```suggestion
  license_file:
   - COPYING
   - additional-licenses-of-revended-packages
```
The logs show that maven is downloading java dependencies for your package. Since you are including external packages in this package, you need to collect the licenses of these packages and include them in the license_file section of the conda recipe. There may be a java package that can do this automatically for you, but I'm not familiar with java build systems, so I don't know.

---

I've modified the build process as to add a step that identifies the license for each dependency, reports which license applies to which dependency (this is saved in file licenses/THIRD-PARTY.txt), and downloads under licenses/dependencies/ the license file for each license. This way, all license files are available when maven finishes.